pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -58,29 +58,46 @@ const (
 
 
 func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
```

Comment:
If there are multiple parallel calls to DiscoverNode method for the same node, we end up discovering multiple times. We can avoid that if we acquire a lock per node. We discussed that this could happen if multiple threads are rediscovering the same node at the same time. Could you please file an issue to address this optimization? This is not urgent though.